### PR TITLE
feat: upgrading etcd version in helm chart dependency for apisix

### LINKED ISSUE #506: etcd dependency upgrade
Hello everybody 👋

We are currently having problems with the current APISIX Helm Chart (`1.1.1`) deployment on our kubernetes cluster.
In fact, we are encountering numerous `etcd` errors like that, making `etcd` component instable :

```
{"level":"panic","ts":"2023-03-09T08:17:11.675Z","caller":"etcdserver/server.go:515","msg":"failed to recover v3 backend from snapshot","error":"failed to find database snapshot file (snap: snapshot file doesn't exist)","stacktrace":"go.etcd.io/etcd/server/v3/etcdserver.NewServer\n\t/go/src/go.etcd.io/etcd/release/etcd/server/etcdserver/server.go:515\ngo.etcd.io/etcd/server/v3/embed.StartEtcd\n\t/go/src/go.etcd.io/etcd/release/etcd/server/embed/etcd.go:245\ngo.etcd.io/etcd/server/v3/etcdmain.startEtcd\n\t/go/src/go.etcd.io/etcd/release/etcd/server/etcdmain/etcd.go:228\ngo.etcd.io/etcd/server/v3/etcdmain.startEtcdOrProxyV2\n\t/go/src/go.etcd.io/etcd/release/etcd/server/etcdmain/etcd.go:123\ngo.etcd.io/etcd/server/v3/etcdmain.Main\n\t/go/src/go.etcd.io/etcd/release/etcd/server/etcdmain/main.go:40\nmain.main\n\t/go/src/go.etcd.io/etcd/release/etcd/server/main.go:32\nruntime.main\n\t/go/gos/go1.16.15/src/runtime/proc.go:225"}
panic: failed to recover v3 backend from snapshot
```

An Issue was raised on `etcd` Repository in late 2022 : https://github.com/etcd-io/etcd/issues/14749
The error was fixed in `etcd` in version `>= 3.5.6`.

I made several tests upgrading APISIX Helm Chart `etcd` dependency to the latest `bitnami/etcd` Helm Chart version (`8.7.7`), and errors disappeared. `etcd` service is now stable.

Can you consider updating the `etcd` dependency in current APISIX Helm Chart ?

### PR DESCRIPTION
**Pull request by heresie**
Hello, 

Closes #506 

All comments are welcome.

### CODE REVIEW

**Review by Gallardot — APPROVED** [review]
LGTM! Thanks for your contributions!

File: charts/apisix/Chart.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 1.2.0
+version: 1.2.1
```
Comment (tao12345666333) [inline charts/apisix/Chart.yaml:34]: Due to the large span of etcd version upgrades, I tend to upgrade a large version, how about we upgrade the version number to 1.3.0?
```suggestion
version: 1.3.0
```

Comment (heresie) [reply charts/apisix/Chart.yaml]: Hello @tao12345666333 

Thank you for the feedback. I updated the Chart version to `1.3.0`.

**Review by tao12345666333 — APPROVED** [review]
LGTM 

Let's wait for CI

**Pull request merged**